provider/datagen/src/transform/segmenter/mod.rs

```diff
@@ -267,6 +119,155 @@ impl crate::DatagenProvider {
         let data = sets::load_extended_pictographic(self).expect("The data should be valid!");
         let extended_pictographic = data.as_borrowed();
 
+        fn set_break_state(
+            break_state_table: &mut [i8],
+            property_length: usize,
+            left_index: usize,
+            right_index: usize,
+            break_state: i8,
+        ) {
+            let index = left_index * property_length + right_index;
+            if break_state_table[index] == UNKNOWN_RULE
+                || break_state_table[index] == NOT_MATCH_RULE
+            {
+                break_state_table[index] = break_state;
+            }
+        }
+
+        fn get_index_from_name(properties_names: &[String], s: &str) -> Option<usize> {
+            properties_names.iter().position(|n| n.eq(s))
+        }
+
+        fn get_word_segmenter_value_from_name(name: &str) -> WordBreak {
+            match name {
+                "ALetter" => WordBreak::ALetter,
+                "CR" => WordBreak::CR,
+                "Double_Quote" => WordBreak::DoubleQuote,
+                "Extend" => WordBreak::Extend,
+                "ExtendNumLet" => WordBreak::ExtendNumLet,
+                "Format" => WordBreak::Format,
+                "Katakana" => WordBreak::Katakana,
+                "Hebrew_Letter" => WordBreak::HebrewLetter,
+                "LF" => WordBreak::LF,
+                "MidLetter" => WordBreak::MidLetter,
+                "MidNum" => WordBreak::MidNum,
+                "MidNumLet" => WordBreak::MidNumLet,
+                "Newline" => WordBreak::Newline,
+                "Numeric" => WordBreak::Numeric,
+                "Regional_Indicator" => WordBreak::RegionalIndicator,
+                "Single_Quote" => WordBreak::SingleQuote,
+                "WSegSpace" => WordBreak::WSegSpace,
+                "ZWJ" => WordBreak::ZWJ,
+                _ => {
+                    panic!("Invalid property name")
+                }
+            }
+        }
+
+        fn get_grapheme_segmenter_value_from_name(name: &str) -> GraphemeClusterBreak {
```

Comment:
Issue: This is a complicated change and it is more difficult to review when you move around code that is unrelated to the change. It appears that you changed this function from being a module function to being an associated function.